tests/test_products.py

```diff
@@ -982,3 +982,73 @@ def test_product_fail_for_hyper_raster(self):
         self.assertTrue('blue_enhanced' in str(ex.exception))
         self.assertTrue('red_enhanced' in str(ex.exception))
         self.assertTrue('green_enhanced' in str(ex.exception))
+
+
+class TestCustomProduct(unittest.TestCase):
+    def test_custom_red_plus_1(self):
+
+        def foo(red):
+            arr = red + 1
+            return arr
+
+        custom_product = ProductsFactory.get_object('custom', function=foo, required_bands=['red'],
```

Comment:
in the case we have only a single band, we can ease the usage with lambda expressions

Reply:
yes, but still you will need to set `require_bands` and `output_bands`